src/test/java/io/jenkins/plugins/audit/listeners/UserCreationListenerTest.java

```diff
@@ -0,0 +1,161 @@
+package io.jenkins.plugins.audit.listeners;
+
+import java.util.List;
+import java.util.HashMap;
+import java.io.IOException;
+import java.lang.reflect.Field;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.jvnet.hudson.test.Issue;
+import org.xml.sax.SAXException;
+import org.acegisecurity.Authentication;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.apache.logging.log4j.core.LogEvent;
+import org.jvnet.hudson.test.JenkinsRule.WebClient;
+import org.apache.logging.log4j.test.appender.ListAppender;
+import org.apache.logging.log4j.message.StructuredDataMessage;
+
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import static org.junit.Assert.*;
+import static org.hamcrest.Matchers.containsString;
+
+import hudson.model.User;
+import jenkins.model.Jenkins;
+import junitparams.Parameters;
+import junitparams.JUnitParamsRunner;
+import hudson.security.pages.SignupPage;
+import hudson.security.HudsonPrivateSecurityRealm;
+
+
+@RunWith(JUnitParamsRunner.class)
+public class UserCreationListenerTest {
+
+    private ListAppender app;
+    private WebClient client;
+    private final HashMap<String, String> USERS = new HashMap<String, String>();
+
+    private static void assertEventCount(final List<LogEvent> events, final int expected) {
+        assertEquals("Incorrect number of events.", expected, events.size());
```

Comment:
If you're interested, you can look into Hamcrest (or AssertJ if you're feeling adventurous) to turn this into a fluent assertion.

Reply:
will give it a try in future tasks :)